src/library/scala/collection/mutable/OpenHashMap.scala

```diff
@@ -136,7 +149,11 @@ extends AbstractMap[Key, Value]
       None
     } else {
       val res = entry.value
```

Comment:
It's somewhat contentious whether or not the `entry.key` should also be updated at this point.

I would say yes, but do the change in a separate PR.